langchain4j-neo4j/src/main/java/dev/langchain4j/store/embedding/neo4j/Neo4jEmbeddingStore.java

```diff
@@ -0,0 +1,244 @@
+package dev.langchain4j.store.embedding.neo4j;
+
+import dev.langchain4j.data.embedding.Embedding;
+import dev.langchain4j.data.segment.TextSegment;
+import dev.langchain4j.store.embedding.EmbeddingMatch;
+import dev.langchain4j.store.embedding.EmbeddingStore;
+import lombok.Builder;
+import lombok.Getter;
+import org.neo4j.driver.AuthTokens;
+import org.neo4j.driver.GraphDatabase;
+import org.neo4j.driver.Session;
+import org.neo4j.driver.Values;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.*;
+import java.util.stream.IntStream;
+
+import static dev.langchain4j.internal.Utils.getOrDefault;
+import static dev.langchain4j.internal.Utils.isCollectionEmpty;
+import static dev.langchain4j.internal.Utils.randomUUID;
+import static dev.langchain4j.internal.ValidationUtils.*;
+import static java.util.Collections.singletonList;
+
+import org.neo4j.driver.Driver;
+import org.neo4j.driver.SessionConfig;
+
+import static dev.langchain4j.store.embedding.neo4j.Neo4jEmbeddingUtils.*;
+
+/**
+ * Represents a Vector index as an embedding store.
+ * Annotated with `@Getter` to be used in {@link Neo4jEmbeddingUtils}
+ */
+@Getter
+public class Neo4jEmbeddingStore implements EmbeddingStore<TextSegment> {
```

Comment:
is there any API in langchain4j to allow existing data in the database to be embedded and indexed?
what does the whole roundtrip look like in an example?